### DISCUSSION TITLE
Updating from a background thread?

### DISCUSSION THREAD

**Discussion by jaredkeithwhite** [a-n-t-h-o-n-y/TermOx#55] [Q&A]
Hi there,

I am interested in building out a GUI using this framework.  My application will have a few background threads that read from various types of sockets and spit out information on the screen accordingly.

I've been going through the examples and they look fairly straightforward as far as GUI layout goes, and even user interactivity.  But how can I prompt the refresh of the panels when my thread gets some event that it wants to prompt an update by?

**Comment by vl-01** [comment]
Call `ox::System::post_event(Custom_event{[&]{ callYourUpdateMethodHere(); }});` from any thread. The code inside the lambda will be executed in the GUI thread. And in your update method, don't forget to call `.update()` on your widget to trigger a redraw after you have updated your app state.
